docs/Required-Link-Flags.md

```diff
@@ -1,38 +1,172 @@
 
-The following linker flags need to be specified when building an application that includes J2ObjC-generated source files:
-````
--L $(j2objc_distribution)/lib -l jre_emul -ObjC
-````
+The link build step (Xcode's "Link Binary With Libraries" build phase) needs J2ObjC-specific flags,
+which vary depending on how your application uses translated Java classes. The core flags are set by
+the [j2objcc](j2objcc.html) command-line script, but need to be specified when building with Xcode.
 
-### **-L $(j2objc_distribution)/lib**
+## Library Search Path
 
-The "$(j2objc_distribution)" path above is the path to your local copy of J2ObjC.  For example, if you unzipped a [J2ObjC release archive file](https://github.com/google/j2objc/releases) to "/usr/local/", this path would be "/usr/local/j2objc" **Important**: do not actually use "$(j2objc_distribution)" in your project; always specify the actual path where you installed J2ObjC.
+J2ObjC's distribution includes several static libraries; to use them, your project needs to tell
+the linker where to find them.
 
-If you build J2ObjC from a copy of its source code, then "$(j2objc_distribution)" is your copy's "j2objc/dist/" directory. This directory will not exist until you build J2ObjC with `make` or `make dist`.
+Generally, the library search path needs to include "$(j2objc_distribution)/lib", where the
+"$(j2objc_distribution)" variable is the path to your local copy of J2ObjC.  For example, if you
+unzipped a [J2ObjC release archive file](https://github.com/google/j2objc/releases) to 
+"/usr/local/", this path would be "/usr/local/j2objc".
 
-### **-force_load $(j2objc_distribution)/lib/libjre_emul.a**
+**Important**: Do not actually use "$(j2objc_distribution)" in your project; always specify the
+actual path where you installed J2ObjC.
 
-The ensures that the JRE emulation library's Objective-C categories are always loaded. Without this flag, some apps with fail with runtime errors that NSStrings do not respond to some selectors, because those selectors are defined by the JRE library's NSString+!JavaString category.
+If you build J2ObjC from a copy of its source code, then "$(j2objc_distribution)" is your copy's 
+"j2objc/dist/" directory. This directory will not exist until you build J2ObjC with `make dist`.
 
-### **-l jre_emul**
+### Xcode: Library Search Paths
 
-This specifies J2ObjC's JRE emulation library, referenced by all generated source files.
+Update the app target's Library Search Paths by adding "$(j2objc_distribution)/lib" (again, use
+the real path).
 
-### **-ObjC**
+## JRE Libraries
 
-This flag tells the linker to include all Objective-C categories in your application. J2ObjC requires this flag because it has categories that extend NSObject and NSString to be support `java.lang.Object` and `java.lang.String` functionality.
+These libraries implement classes defined by J2ObjC's JRE emulation:
 
-**Note**: if you get "unrecognized selector" link errors, this flag wasn't set.
+<table>
```

Comment:
Markdown provides table syntax, which I use in Translation-Reference.md (https://raw.githubusercontent.com/google/j2objc/gh-pages/docs/Translation-Reference.md). It looks like it will generate the same HTML, but is easier to read in text.